api/src/util.rs

```diff
@@ -31,4 +31,16 @@ mod tests {
         (&conn as &PgConnection).begin_test_transaction().unwrap();
         conn
     }
+
+    /// Assert that the first value is an Err, and that its string form matches
+    /// the second argument.
+    #[macro_export]
+    macro_rules! assert_err {
+        ($res:expr, $msg:tt $(,)?) => {
+            match $res {
+                Ok(_) => panic!("Expected Err, got Ok"),
+                Err(err) => assert_eq!(format!("{}", err), $msg),
```

Comment:
is there a generic `fail!` or something like that so we dont have to panic?

Reply:
That's exactly what `panic!` is. It indicates an internal bug in a pretty unlikely scenario.